pkg/services/sqlstore/permissions/dashboard.go

```diff
@@ -148,7 +148,7 @@ func (f *accessControlDashboardPermissionFilter) buildClauses() {
 
 		if len(toCheck) > 0 {
 			if !useSelfContainedPermissions {
-				builder.WriteString("(dashboard.uid IN (SELECT substr(scope, 16) FROM permission WHERE scope LIKE 'dashboards:uid:%'")
+				builder.WriteString("(dashboard.uid IN (SELECT identifier FROM permission WHERE kind = 'dashboards'")
```

Comment:
Thought: I wonder if we should also add `AND attribute = 'uid'` to this and the other queries. It should always be `uid` for dashboards, but just in case.